app/src/main/java/org/ea/sqrl/utils/Utils.java

```diff
@@ -76,4 +79,20 @@ public static void refreshStorageFromDb(Activity activity) throws Exception {
         SQRLStorage sqrlStorage = SQRLStorage.getInstance(activity);
         sqrlStorage.read(identityData);
     }
+
+    public static byte[] getFileIntentContent(Context context, Uri contentUri) {
+        if (contentUri == null) return null;
+
+        try {
+            InputStream inputStream = context.getContentResolver().openInputStream(contentUri);
+            byte[] data = new byte[inputStream.available()];
```

Comment:
"Note that while some implementations of InputStream will return the total number of bytes in the stream, many will not. It is never correct to use the return value of this method to allocate a buffer intended to hold all data in this stream."

https://docs.oracle.com/javase/7/docs/api/java/io/InputStream.html#available()

Can you use a ByteBuffer or a ByteArrayOutputStream instead so we can ensure that all bytes are read?

Reply:
Correct. I read that `InputStream.available()` will return the full number of bytes if reading from a file off of local storage, but the docs make it absolutely clear that this is never a correct implementation.   I will fix this asap.

Reply:
Should be fixed with latest commit.